src/image.rs

```diff
@@ -0,0 +1,52 @@
+use std::path::Path;
+
+use crate::file::functions::read_file_from_image;
+use crate::file::functions::Partition;
+use anyhow::{Context, Result};
+use regex::Regex;
+
+// NOTE (2024-05-29 Tobias Langer): /etc/os-release is a symlink in our yocto
+// builds. The e2tools-suite cannot handle symlinks so we use its target
+// directly.
+const OS_RELEASE_PATH: &str = "/usr/lib/os-release";
+const OS_RELEASE_PARTITION: Partition = Partition::rootA;
+
+lazy_static::lazy_static! {
+    pub static ref ARCH_REGEX: Regex = {
+        Regex::new(r#"OMNECT_TARGET_ARCH="(?<arch>.*)""#).unwrap()
+    };
+}
+
+pub enum Architecture {
+    Arm64,
+}
+
+impl TryInto<Architecture> for &str {
+    type Error = anyhow::Error;
+
+    fn try_into(self) -> Result<Architecture> {
+        let arch = match self {
+            "aarch64" => Architecture::Arm64,
```

Comment:
what about amd64?

Reply:
any other architecture that we're supporting?